File: 1480.RunningSumOf1dArray.py
# class Solution(object):
#     def runningSum(self, nums):
#         result = []

#         for i in range(len(nums)):
#             temp = 0
#             for j in range(i, -1, -1):
#                 temp += nums[j]
#             result.append(temp)
#         return result


# solution = Solution()

# print(solution.runningSum([3, 1, 2, 10, 1]))


class Solution(object):
    def runningSum(self, nums):
        result = 0

        for i in range(len(nums)):
            result = result + nums[i]
            nums[i] = result
        return nums


solution = Solution()

print(solution.runningSum([3, 1, 2, 10, 1]))

File: 1480.RunningSumOf1dArray.js
var runningSum = function (nums) {
  let result = [];

  for (let i = 0; i < nums.length; i++) {
    let temp = 0;
    for (let j = i; j >= 0; j--) {
      temp += nums[j];
    }
    result.push(temp);
  }
  return result;
};

runningSum([3, 1, 2, 10, 1]);
